testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -320,7 +322,7 @@ impl RewardSetEventPayload {
 }
 
 #[cfg(test)]
-static TEST_EVENT_OBSERVER_SKIP_RETRY: std::sync::Mutex<Option<bool>> = std::sync::Mutex::new(None);
+thread_local! { static TEST_EVENT_OBSERVER_SKIP_RETRY: RefCell<bool> = RefCell::new(false); }
```

Comment:
Is it possible to use `Cell` here instead of `RefCell`?

Reply:
Also can you add doc comment to this variable? Skip retry of what? I'm not familiar with this code so I don't know

Reply:
Also, why does it have to be thread-local? If it doesn't have to be thread-local, can we just use the `TestFlag` struct?

Reply:
@hstove tests are executed in parallel so there is a race condition between the set and clear of TEST_EVENT_OBSERVER_SKIP_RETRY (this caused the 3 related tests to often fail when executed togheter)

Reply:
Ah that makes sense, thanks!